classes/issue/Collector.inc.php

```diff
@@ -0,0 +1,429 @@
+    public ?array $numbers = null;
+
+    /** @var array|null return issues with year(s) */
+    public ?array $years = null;
+
+    /** @var array|null return issues that match a title */
+    public ?array $titles = null;
+
+    /** @var string|null Returns Issue by URL path  */
+    public ?string $urlPath = null;
+
+    /** @var string return issues which match words from this search phrase */
+    public string $searchPhrase = '';
+
+    private ?string $_orderByConstant = null;
+
+    public function __construct(DAO $dao)
+    {
+        $this->dao = $dao;
+    }
+
+    /**
+     * Set context issues filter
+     *
+     * @return $this
+     */
+    public function filterByContextIds(?array $contextIds): self
+    {
+        $this->contextIds = $contextIds;
+        return $this;
+    }
+
+    /**
+     * Set result order column
+     *
+     * @return $this
+     */
+    public function orderBy(string $column): self
+    {
+        // Saving this state, so we can conditionally set the raw query for sorting by current issue
+        $this->_orderByConstant = $column;
+
+        switch ($column) {
+            case self::ORDERBY_LAST_MODIFIED:
+                $this->resultOrderings = [
+                    [
+                        'orderBy' => 'i.last_modified',
+                        'direction' => self::ORDER_DIR_DESC
+                    ]
+                ];
+                break;
+            case self::ORDERBY_SEQUENCE:
+                $this->resultOrderings = [
+                    [
+                        'orderBy' => 'o.seq',
+                        'direction' => self::ORDER_DIR_ASC
+                    ]
+                ];
+                break;
+            case self::ORDERBY_PUBLISHED_ISSUES:
+                $this->resultOrderings = [
+                    [
+                        'orderBy' => 'o.seq',
+                        'direction' => self::ORDER_DIR_ASC
+                    ],
+                    [
+                        'orderBy' => 'currentIssue',
+                        'direction' => self::ORDER_DIR_DESC
+                    ],
+                    [
+                        'orderBy' => 'i.date_published',
+                        'direction' => self::ORDER_DIR_DESC
+                    ]
+                ];
+                break;
+            case self::ORDERBY_UNPUBLISHED_ISSUES:
+                $this->resultOrderings = [
+                    [
+                        'orderBy' => 'i.year',
+                        'direction' => self::ORDER_DIR_ASC
+                    ],
+                    [
+                        'orderBy' => 'i.volume',
+                        'direction' => self::ORDER_DIR_ASC
+                    ],
+                    [
+                        'orderBy' => 'i.number',
+                        'direction' => self::ORDER_DIR_ASC
+                    ]
+                ];
+                break;
+            case self::ORDERBY_GATEWAY_LOCKSS:
```

Comment:
This isn't necessary for the PR, but I wanted to add a comment about how plugins can augment the collectors without modifying their code. I suspect the code that calls this sits in the LOCKSS plugin and looks like this:

```php
$collector = Repo::issue()->getCollector();
$collector->orderBy($collector::ORDERBY_SHELF);
$issues = Repo::issue()->getMany($collector);
```

If a plugin needs to do some custom handling here, there are two options. The first is to use the hook to extend the query. That would look like:

```php
$collector = Repo::issue()->getCollector();
$collector->orderBy = 'customOrder';
```

With a hook like:

```php
HookRegistry::register('Issue::Collector', function($hookName, $args) {
    $q = $args[0];
    $collector = $args[1];

    if ($collector->orderBy === 'customOrder') {
        $q->orderBy(...); // Whatever you want.
    }
});
```

However, this is only necessary when you want the custom ordering to be available for for any other code -- for example if another plugin depends on this one, or if this plugin calls this code a lot and you want to standardize it. 

When a plugin only needs to change the query builder for a one-off use case, it can get the Laravel query builder instance and use it's methods directly:

```php
$collector = Repo::issue()->getCollector();
$q = $collector->getQueryBuilder();

// Use Laravel's query builder to set an order and get rows
$q->orderBy(...);
$rows = $q->get();

// Use Laravel's collection methods with the DAO to get objects
$issues = $rows->map(function($row) {
    return Repo::issue()->dao->fromRow($row);
});
```

I think that this kind of one-off usage is ok, especially in plugins, when we don't necessarily want this custom result to be available to everyone.